lib/drivers/tandemTslimDriver.js

```diff
@@ -1374,6 +1375,26 @@ module.exports = function (config) {
     return records;
   };
 
+  var buildNewDayRecords = function(data, records) {
+    var newDayRecords = filterLogEntries([PUMP_LOG_RECORDS.LID_NEW_DAY], data.log_records);
+
+    for (var b = 0; b < newDayRecords.length; ++b) {
+      var event = newDayRecords[b];
+      var rate = event.commanded_basal_rate;
+      if ((rate !== undefined) && (rate > 0)) {
```

Comment:
Possibly an even safer check might be `rate != null` since this covers both `null` and `undefined`.